src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -377,3 +391,56 @@ SupportFunctionForColumn(Var *partitionColumn, Oid accessMethodId,
 
 	return supportFunctionOid;
 }
+
+
+/*
+ * TableHasData function checks whether given undistributed table contains any row and
+ * returns true if there is any data. This function is only for undistributed tables and
```

Comment:
Instead of saying should not be called, we may probably want to say it'd error out if used with distributed tables and explicitly error for distributed tables. (an d possibly change the name of the function to `LocalTableEmpty()`)